### PR TITLE
Add support for multi platform builds

### LINKED ISSUE #123: ARM support
Hello.

This operator doesn´t seem to support ARM processors as I got an `exec format error` when starting the container.

It would be nice to be able to use this on ARM processors.

It should be relatively simple to do it with Docker and go: https://docs.docker.com/build/building/multi-platform/

### PR DESCRIPTION
**Pull request by brpaz**
## Context

ARM-based platforms are getting more and more popular, and most well know projects and helm charts already support ARM architectures. (Ex: Prometheus, Grafana, Postgres, Cert Manager, etc). The only thing that I need in my Kubernetes cluster that was missing support for ARM was this operator.

This PR aims to fix that.  Closes #123 

## Changes

- **Dockerfile** -  Add support for multi-platform builds following recommendations from [this](https://docs.docker.com/build/building/multi-platform/) and [this](https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/).  Also refactored a bit the Dockerfile with cache support, which is important to optimize the build time, when building for multiple platforms.
- **GitHub Actions** - Refactored the release workflow to leverage, official docker GitHub actions, which makes building images for multi-platform easier. Reference: https://docs.docker.com/build/ci/github-actions/multi-platform/

### CODE REVIEW

File: build/Dockerfile.dist [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
-COPY . /go/src/github.com/movetokube/postgres-operator
-WORKDIR /go/src/github.com/movetokube/postgres-operator/cmd/manager
+ARG TARGETPLATFORM
+ARG BUILDPLATFORM
+ARG TARGETOS
+ARG TARGETARCH
 
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o /usr/local/bin/postgres-operator
+WORKDIR /src
+COPY go.mod go.sum ./
```
Comment (hitman99) [inline build/Dockerfile.dist:10]: why not just `COPY . .` and then run `go mod download` ?

**Review by hitman99 — COMMENTED** [review]
Great PR, thank you for the contribution

File: build/Dockerfile.dist (hunk 1 above)
Comment (reddec) [reply build/Dockerfile.dist:10]: I suppose for caching purpose https://docs.docker.com/language/golang/build-images/

Comment (hitman99) [reply build/Dockerfile.dist:10]: I see, ok

**Review by hitman99 — APPROVED** [review]

**Pull request merged**